### PR TITLE
Add lifecycle rules for parameters and responses 

### PR DESCRIPTION
**Pull request by acunniffe**
A PR of the latest we've completed, today's batch, all about how parameters can change between versions. The `.changed` functions really shine here

### CODE REVIEW

**Review by cmars — COMMENTED** [review]
A few minor things below.

Also, Snyk found a minor security issue in one of the package dependencies. These aren't show stoppers, but if possible, can we use `json-ptr@3.0.0`?

Screen Shot 2021-11-10 at 2 55 13 PM

**Comment by cmars** [comment]
Working on getting this feature branch integrated into CircleCI, but I'm also seeing test failures when I run `yarn test` on this branch.

File: src/rulesets/tests/operations.test.ts [hunk 1]
```diff
@@ -231,6 +222,106 @@ describe("operation parameters", () => {
       expect(result.results[0].passed).toBeTruthy();
       expect(result).toMatchSnapshot();
     });
+
+    it("fails when adding a required query parameter", async () => {
+      // const base = JSON.parse(JSON.stringify(baseForOperationMetadataTests));
+      // base.paths!["/example"]!.get!.parameters = [];
```
[the remark this answers is not in the document]
Comment (acunniffe) [reply src/rulesets/tests/operations.test.ts]: Yes. done

**Comment by acunniffe** [comment]
> Working on getting this feature branch integrated into CircleCI, but I'm also seeing test failures when I run `yarn test` on this branch.

@cmars I am not getting errors locally, and CI appears to pass. Are you able to share a screenshot, the error or a GitHub action log?

**Comment by cmars** [comment]
> > Working on getting this feature branch integrated into CircleCI, but I'm also seeing test failures when I run `yarn test` on this branch.
> 
> @cmars I am not getting errors locally, and CI appears to pass. Are you able to share a screenshot, the error or a GitHub action log?

Seems to be passing now with your latest push. Landing now!

**Pull request merged**